### PR TITLE
V3-805 Extend satellitedb with Project members entity

### PR DESCRIPTION
**Pull request by crawter**

### CODE REVIEW

File: pkg/satellite/projectmembers.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	// GetByMemberID is a method for querying project member from the database by memberID.
+	GetByMemberID(ctx context.Context, memberID uuid.UUID) (*ProjectMember, error)
+	// GetByProjectID is a method for querying project members from the database by projectID.
+	GetByProjectID(ctx context.Context, projectID uuid.UUID) ([]ProjectMember, error)
+	// Get is a method for querying project member from the database by id.
+	Get(ctx context.Context, id uuid.UUID) (*ProjectMember, error)
+	// Insert is a method for inserting project member into the database.
+	Insert(ctx context.Context, memberID, projectID uuid.UUID) (*ProjectMember, error)
+	// Delete is a method for deleting project member by Id from the database.
+	Delete(ctx context.Context, id uuid.UUID) error
```
Comment (egonelbre) [inline pkg/satellite/projectmembers.go]: `Remove` seems more appropriate.

File: pkg/satellite/projectmembers.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	// Update is a method for updating project member entity.
+	Update(ctx context.Context, projectMember *ProjectMember) error
+}
+
+// ProjectMember is a database object that describes ProjectMember entity.
+type ProjectMember struct {
+	ID uuid.UUID
```
Comment (egonelbre) [inline pkg/satellite/projectmembers.go]: Cannot we use `MemberID` and `ProjectID` as a key instead?

File: pkg/satellite/satellitedb/projectmembers.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	projectID, err := bytesToUUID(projectMember.ProjectId)
+	if err != nil {
+		return nil, err
+	}
+
+	u := &satellite.ProjectMember{
```
Comment (egonelbre) [inline pkg/satellite/satellitedb/projectmembers.go]: nit, `return &satellite.ProjectMember{ ... }, nil`, makes it slightly more clear that this is the final result.

File: pkg/satellite/satellitedb/projectmembers_test.go [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+	db, err := New("sqlite3", "file::memory:?mode=memory&cache=shared")
+	if err != nil {
+		assert.NoError(t, err)
+	}
+	defer ctx.Check(db.Close)
+
+	// creating tables
+	err = db.CreateTables()
+	if err != nil {
+		assert.NoError(t, err)
```
Comment (egonelbre) [inline pkg/satellite/satellitedb/projectmembers_test.go]: Use `t.Fatal(err)` here, it shouldn't continue when this fails.

File: pkg/satellite/satellitedb/projectmembers_test.go [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+)
+
+func TestProjectMembersRepository(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	// creating in-memory db and opening connection
+	db, err := New("sqlite3", "file::memory:?mode=memory&cache=shared")
+	if err != nil {
+		assert.NoError(t, err)
```
Comment (egonelbre) [inline pkg/satellite/satellitedb/projectmembers_test.go]: Use `t.Fatal(err)` here, it shouldn't continue when this fails.

File: pkg/satellite/satellitedb/projectmembers_test.go [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+package satellitedb
+
+import (
+	"context"
+	"testing"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/stretchr/testify/assert"
+	"storj.io/storj/internal/testcontext"
+	stlt "storj.io/storj/pkg/satellite"
```
Comment (egonelbre) [inline pkg/satellite/satellitedb/projectmembers_test.go]: Don't alias the `satellite`.

File: pkg/satellite/satellitedb/projectmembers_test.go [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+		members, err = projectMembers.GetByProjectID(ctx, createdProjects[0].ID)
+		assert.NotNil(t, members)
+		assert.Equal(t, 2, len(members))
+		assert.Nil(t, err)
+		assert.NoError(t, err)
+	})
+}
+
+func prepareUsersAndProjects(ctx context.Context, users stlt.Users, projects stlt.Projects) ([]stlt.User, []stlt.Project) {
+	user1 := &stlt.User{
```
Comment (egonelbre) [inline pkg/satellite/satellitedb/projectmembers_test.go]: You can use here:
```
users := []*satellite.User{{
		Email:        "email2@ukr.net",
		PasswordHash: []byte("some_readable_hash"),
		LastName:     "LastName",
		FirstName:    "FirstName",
	}, {
		Email:        "email2@ukr.net",
		PasswordHash: []byte("some_readable_hash"),
		LastName:     "LastName",
		FirstName:    "FirstName",
	}, ...
}

var err error
for i, user := range users {
    users[i], err = users.Insert(ctx, user)
    if err != nil {
        t.Fatal(err)
    }
}
```

File: pkg/satellite/satellitedb/projectmembers_test.go [hunk 8]
```diff
@@ -186,45 +186,54 @@ func TestProjectMembersRepository(t *testing.T) {
...
-	project2, _ = projects.Insert(ctx, project2)
+	for i, project := range projectList {
+		projectList[i], err = projects.Insert(ctx, project)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
 
-	return []stlt.User{*user1, *user2, *user3}, []stlt.Project{*project1, *project2}
+	return []stlt.User{*usersList[0], *usersList[1], *usersList[2]}, []stlt.Project{*projectList[0], *projectList[1]}
```
Comment (egonelbre) [inline pkg/satellite/satellitedb/projectmembers_test.go]: `return usersList, projectList`

Comment (crawter) [reply pkg/satellite/satellitedb/projectmembers_test.go]: omg, i should take a break :D 
i'm too inattentive

File: pkg/satellite/satellitedb/projectmembers_test.go (hunk 4 above)
Comment (kaloyan-raev) [reply pkg/satellite/satellitedb/projectmembers_test.go]: Another way would be:
```go
if !assert.NoError(t, err) {
    return
}
```

File: pkg/satellite/projectmembers.go (hunk 2 above)
Comment (coyle) [reply pkg/satellite/projectmembers.go]: It seems like it would be odd to call ProjectMember.ProjectID instead of ProjectMember.ID

**Pull request merged**